src/e2e_test/perf_tool/pkg/run/run.go

```diff
@@ -108,6 +112,43 @@ func (r *Runner) RunExperiment(ctx context.Context, expID uuid.UUID, spec *exper
 	defer r.clusterCleanup()
 	defer r.clusterCtx.Close()
 
+	var egCtx context.Context
+	r.eg, egCtx = errgroup.WithContext(ctx)
+	r.eg.Go(func() error {
+		return r.runActions(egCtx, spec)
+	})
```

Comment:
maybe write a small note here saying that you are relying on errgroup cancellation of context on first error.